src/components/OrderBook/OrderBookChart.tsx

```diff
@@ -0,0 +1,264 @@
+  .amcharts-Button,
+  .amcharts-AxisLabel-group,
+  .amcharts-Axis,
+  .amcharts-ValueAxis,
+  .amcharts-AxisTitles {
+    fill-opacity: 0.8;
+    tspan {
+      fill: var(--color-text-primary);
+    }
+  }
+  .amcharts-Button-group {
+    cursor: pointer;
+    tspan {
+      font-weight: 800;
+    }
+  }
+`
+
+interface OrderBookErrorProps {
+  error: Error
+}
+
+export const OrderBookError: React.FC<OrderBookErrorProps> = ({ error }) => <Wrapper>{error.message}</Wrapper>
+
+const OrderBookChart: React.FC<OrderBookChartProps> = (props) => {
+  const { baseToken, quoteToken, networkId, data } = props
+  const mountPoint = useRef<HTMLDivElement>(null)
+  const chartRef = useRef<am4charts.XYChart | null>(null)
+
+  const [initialZoom, setInitialZoom] = useSafeState<ZoomValues>({ startX: 0, endX: 1, endY: 1 })
+  const [bids, setBids] = useSafeState<PricePointDetails[]>([])
+  const [asks, setAsks] = useSafeState<PricePointDetails[]>([])
+
+  useEffect(() => {
+    if (!mountPoint.current) return
+    const chart = createChart(mountPoint.current)
+    chartRef.current = chart
+
+    // dispose on mount only
+    return (): void => chart.dispose()
+  }, [])
+
+  useEffect(() => {
+    if (!chartRef.current || data === null) return
+
+    if (data.length === 0) {
+      chartRef.current.data = []
+      return
+    }
+
+    // go on with the update when data is ready
+    drawLabels({
+      chart: chartRef.current,
+      baseToken,
+      quoteToken,
+      networkId,
+    })
+
+    const asks: PricePointDetails[] = []
+    const bids: PricePointDetails[] = []
+
+    for (const pricePoint of data) {
+      if (pricePoint.type === Offer.Bid) {
+        bids.push(pricePoint)
+      } else {
+        asks.push(pricePoint)
+      }
+    }
+
+    setAsks(asks)
+    console.log('asks', asks)
+    setBids(bids)
+    console.log('bids', bids)
+
+    const initialZoom = calcInitialZoom(bids, asks)
+    console.log('initialZoom', initialZoom)
+
+    const {
+      xAxes: [xAxis],
+      yAxes: [yAxis],
+    } = chartRef.current
+
+    // Setting initial zoom
+    xAxis.start = initialZoom.startX
+    xAxis.end = initialZoom.endX
+    yAxis.end = initialZoom.endY
+    // Storing calculated zoom values
+    setInitialZoom(initialZoom)
+
+    chartRef.current.data = data
+  }, [baseToken, networkId, quoteToken, data, setInitialZoom, setAsks, setBids])
+
+  // Creates zoom buttons once initialZoom has been calculated
+  useEffect(() => {
+    // return
+    const chart = chartRef.current
+    if (!chart) {
+      return
+    }
+
+    // Finding the container for zoom buttons
+    const buttonContainer = getZoomButtonContainer(chart)
+
+    // Data not loaded yet, there's no container
+    if (!buttonContainer) {
+      return
+    }
+    buttonContainer.disposeChildren()
+
+    const xAxis = chart.xAxes.values[0] as am4charts.ValueAxis<am4charts.AxisRenderer>
+    const yAxis = chart.yAxes.values[0] as am4charts.ValueAxis<am4charts.AxisRenderer>
```

Comment:
Can use the same trick you added on line 166

```
    const {
      xAxes: [xAxis],
      yAxes: [yAxis],
    } = chartRef.current
```

Reply:
Yes, gonna use this and a bunch of other tricks. For now this is your commits adapted to work with different fetch logic and other refactors that have cropped up since.